yaml.go

```diff
@@ -141,6 +152,20 @@ func (dec *Decoder) Decode(v interface{}) (err error) {
 // conversion of YAML into a Go value.
 func (n *Node) Decode(v interface{}) (err error) {
 	d := newDecoder()
+	return n.decode(d, v)
+}
+
+// DecodeWithConfig decodes the node and stores its data into the value pointed to by v.
+// It uses passed DecodeConfig to setup internal decoder.
+func (n *Node) DecodeWithConfig(v interface{}, config *DecodeConfig) (err error) {
```

Comment:
If we invert the order of fields, we might name it as just _DecodeWith_, which would make it read nicely as in:
```
err := node.DecodeWith(options, &value)
```

We should probably also have other functions and methods taking options, but it's okay to start with just these.

Reply:
This is done: `yaml.UnmarshalWith`, `yaml.NewDecoderWith` and `node.DecodeWith` are added.